avocado/core/job.py

```diff
@@ -483,7 +483,7 @@ def _run(self):
 
         if 'INTERRUPTED' in summary:
             self.exitcode |= exit_codes.AVOCADO_JOB_INTERRUPTED
-        if 'FAIL' in summary:
+        if 'FAIL' in summary and not getattr(self.args, 'relax_exit_code', False):
```

Comment:
Not difference except the fact that second variant requires shell, which is sometimes not possible